tests/runner/testresource.go

```diff
@@ -90,26 +90,60 @@ func (r *TestResources) FindActiveResource(name string) Disposable {
 func (r *TestResources) setup() error {
 	r.ctx, r.cancel = context.WithCancel(context.Background())
 
-	for dr := r.dequeueResource(); dr != nil; dr = r.dequeueResource() {
-		err := dr.Init(r.ctx)
-		r.pushActiveResource(dr)
+	resourceCount := 0
+	errs := make(chan error)
+	for {
+		dr := r.dequeueResource()
+		if dr == nil {
+			break
+		}
+
+		resourceCount++
+		go func() {
+			err := dr.Init(r.ctx)
+			r.pushActiveResource(dr)
+			errs <- err
+		}()
+	}
+
+	for i := 0; i < resourceCount; i++ {
+		err := <-errs
 		if err != nil {
 			return err
 		}
 	}
+
 	return nil
```

Comment:
We should `errors.Join` here, instead of returning on the first error.

Reply:
Done